### PR TITLE
Make ClusterQueue's AdmissionCheckStrategy a pointer, avoid type casting

### PR DESCRIPTION
**Pull request by PBundyra**
#### What type of PR is this?
/kind bug

#### What this PR does / why we need it:
Addresses comments of an already merged PR #1960 

#### Which issue(s) this PR fixes:

Part of #1432 

#### Does this PR introduce a user-facing change?

```release-note
NONE

```

### CODE REVIEW

File: pkg/webhooks/clusterqueue_webhook.go [hunk 1]
```diff
@@ -127,7 +127,7 @@ func validatePreemption(preemption *kueue.ClusterQueuePreemption, path *field.Pa
 
 func validateCQAdmissionChecks(spec *kueue.ClusterQueueSpec, path *field.Path) field.ErrorList {
 	var allErrs field.ErrorList
-	if len(spec.AdmissionChecksStrategy.AdmissionChecks) != 0 && len(spec.AdmissionChecks) != 0 {
+	if spec.AdmissionChecksStrategy != nil && len(spec.AdmissionChecks) != 0 {
```
Comment (tenzen-y) [inline pkg/webhooks/clusterqueue_webhook.go:130]: Could you update this unit test like `AdmissionCheckStrategy()`?

https://github.com/kubernetes-sigs/kueue/blob/3112eec330ee88f153558c68636380658bcf917c/pkg/webhooks/clusterqueue_webhook_test.go#L76-L77

File: pkg/util/admissioncheck/admissioncheck.go [hunk 2]
```diff
@@ -157,15 +157,20 @@ func FilterProvReqAnnotations(annotations map[string]string) map[string]string {
...
-			checks[check.Name].Insert(string(flavor))
+func NewAdmissionChecks(cq *kueue.ClusterQueue) map[string]sets.Set[kueue.ResourceFlavorReference] {
+	var checks map[string]sets.Set[kueue.ResourceFlavorReference]
+	if cq.Spec.AdmissionChecksStrategy != nil {
+		checks = make(map[string]sets.Set[kueue.ResourceFlavorReference], len(cq.Spec.AdmissionChecksStrategy.AdmissionChecks))
+		for _, check := range cq.Spec.AdmissionChecksStrategy.AdmissionChecks {
+			checks[check.Name] = sets.New[kueue.ResourceFlavorReference]()
+			for _, flavor := range check.OnFlavors {
+				checks[check.Name].Insert(flavor)
+			}
```
Comment (tenzen-y) [inline pkg/util/admissioncheck/admissioncheck.go:168]:
```suggestion
			checks[check.Name] = sets.New[kueue.ResourceFlavorReference](check.OnFlavors...)
```

**Review by tenzen-y — COMMENTED** [review]
Overall lgtm.

File: pkg/webhooks/clusterqueue_webhook.go (hunk 1 above)
Comment (PBundyra) [reply pkg/webhooks/clusterqueue_webhook.go:130]: Could you clarify please? Do you mean to not include any rules in the `AdmissionCheckStrategy` field?

Comment (tenzen-y) [reply pkg/webhooks/clusterqueue_webhook.go:130]: Yes, that's right. I would like to prove if the webhook error happens regardless of the rules.

**Review by tenzen-y — COMMENTED** [review]
Thank you!
/lgtm
/approve

**Review by tenzen-y — COMMENTED** [review]
Thx!
/lgtm

**Pull request merged**